src/activity_handlers.cpp

```diff
@@ -1956,8 +1956,27 @@ void activity_handlers::train_finish( player_activity *act, player *p )
         add_msg( m_good, _( "You learn %s." ), mastyle.name );
         g->events().send<event_type::learns_martial_art>( p->getID(), ma_id );
         p->add_martialart( mastyle.id );
+    }
+    const spell_id &sp_id = spell_id( act->name );
+    if( sp_id.is_valid() ) {
```

Comment:
now that i'm looking at this the second time, it _might_ be better to make its own static function for this, for future expansion of this activity. (i'd probably do one for the martial arts too but that's out of scope)